transfermethodui/src/androidTest/java/com/hyperwallet/android/ui/transfermethod/insight/BankAccountInsightTest.java

```diff
@@ -0,0 +1,290 @@
+        assertThat(mParamsCaptor.getValue().get("erfd"), is("bankAccountPurpose"));
+    }
+
+    @Test
+    public void testAddTransferMethod_verifyInsightEventCreatedOnApiError() {
+        mMockWebServer.mockResponse().withHttpResponseCode(HTTP_BAD_REQUEST).withBody(sResourceManager
+                .getResourceContent("bank_account_invalid_routing_response.json")).mock();
+
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.branchId)).perform(nestedScrollTo(), replaceText(ROUTING_NUMBER));
+        onView(withId(R.id.bankAccountId)).perform(nestedScrollTo(), replaceText(ACCOUNT_NUMBER));
+        onView(withId(R.id.bankAccountPurpose)).perform(nestedScrollTo(), click());
+        onView(allOf(withId(R.id.select_name), withText("Savings"))).perform(click());
+
+        onView(withId(R.id.add_transfer_method_button)).perform(nestedScrollTo(), click());
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(1)).trackError(any(Context.class),
+                eq(AddTransferMethodActivity.TAG),
+                eq(PageGroups.TRANSFER_METHOD),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("error_type"), is("API"));
+        assertThat(mParamsCaptor.getValue().get("error_code"), is("CONSTRAINT_VIOLATIONS"));
+        assertThat(mParamsCaptor.getValue().get("error_message"), is("Routing Number [211179531] is not valid. "
+                + "Please modify Routing Number to a valid ACH Routing Number of the branch of your bank."));
+        assertThat(mParamsCaptor.getValue().get("erfd"), is("branchId"));
+    }
+
+    @Test
+    public void testAddTransferMethod_verifyInsightEventCreatedOnClickCreateTransferMethodWithNetworkError()
+            throws IOException {
+        mActivityTestRule.launchActivity(null);
+
+        onView(withId(R.id.branchId)).perform(nestedScrollTo(), replaceText(ROUTING_NUMBER));
+        onView(withId(R.id.bankAccountId)).perform(nestedScrollTo(), replaceText(ACCOUNT_NUMBER));
+        onView(withId(R.id.bankAccountPurpose)).perform(nestedScrollTo(), click());
+        onView(allOf(withId(R.id.select_name), withText("Savings"))).perform(click());
+
+        mMockWebServer.getServer().shutdown();
+
+        onView(withId(R.id.add_transfer_method_button)).perform(nestedScrollTo(), click());
+
+        verify(mHyperwalletInsightMockRule.getInsight(),
+                timeout(5000).times(1)).trackError(any(Context.class),
+                eq(AddTransferMethodActivity.TAG),
+                eq(PageGroups.TRANSFER_METHOD),
+                mParamsCaptor.capture());
+
+        assertThat(mParamsCaptor.getValue().get("error_type"), is("CONNECTION"));
+        assertThat(mParamsCaptor.getValue().get("error_code"), is("EC_IO_EXCEPTION"));
+//        assertThat(mParamsCaptor.getValue().get("error_message"),
```

Comment:
@skoong @vinotb-altmtrk   any specific reason to have commented out assertions? I did run the tests with those assertions got 

> Expected: is "Oops... Something went wrong, please try again."
Got: "Oops… Something went wrong, please try again."

can they be fixed? Please also look at the other commented out assertions in this class

Reply:
Updated